File: ljl-blog-web/src/types/project.ts
/** 项目时间轴节点 */
export interface ProjectTimelineItem {
  date: string
  title: string
  description: string
}

/** 项目展示 */
export interface Project {
  id: string
  name: string
  description: string
  summary: string
  techStack: string[]
  cover: string
  gitUrl?: string
  demoUrl?: string
  featured?: boolean
  startDate: string
  endDate?: string | null
  /** 详情页扩展字段 */
  devProcess?: string[]
  screenshots?: string[]
  timeline?: ProjectTimelineItem[]
}

/** 项目查询参数 */
export interface ProjectQueryParams {
  page?: number
  pageSize?: number
  keyword?: string
  featured?: boolean
}

File: ljl-blog-server/src/main/java/com/ljlblogserver/entity/Project.java
package com.ljlblogserver.entity;

import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
public class Project {

    private Long id;
    private String name;
    private String description;
    private String summary;
    private String cover;
    private String gitUrl;
    private String demoUrl;
    private Boolean featured;
    private LocalDate startDate;
    private LocalDate endDate;
    private String techStackJson;
    private String devProcessJson;
    private String screenshotsJson;
    private String timelineJson;
    private LocalDateTime createdAt;
}
